examples/kiva/compiled_path.py

```diff
@@ -33,7 +33,8 @@ def compiled_path():
     gc.set_stroke_color((0, 0, 1, 1))
     gc.draw_path_at_points(locs, path, STROKE)
 
-    file_path = tempfile.mktemp(suffix='.jpg')
+    with tempfile.NamedTemporaryFile(suffix='.jpg', delete=False) as fid:
+        file_path = fid.name
```

Comment:
This won't work. The file is destroyed after the context goes away. You'll instead need to move the `with` block into `Demo._create_component` so that the file can exist during both the call to `compiled_path` and `Image.from_file`.

Reply:
That makes sense.
This would require changing the signature of `compiled_path` correct? Otherwise, if the context manager is in `Demo._create_component`, `compiled_path` would not have access to `fid`/`file_path`.

Also as a tangential question, it seems that rather than using `NamedTemporaryFile` in a context manager, we could use `tempfile.mkstemp` instead, without a context manager?  Also, I see that `Image.from_file` calls `PIL.Image.open(filename)`, but I'm not understanding when the file is intended to be closed/stop existing.  Ref: https://pillow.readthedocs.io/en/stable/reference/open_files.html#image-lifecycle

Reply:
That's a good question. Looking into `Image.from_file`: https://github.com/enthought/enable/blob/582fb46b624ec6372ffe5e23b3b2d53d582b056e/enable/primitives/image.py#L37-L42

We see that the image is immediately converted to a numpy array. It should be safe to destroy the file after `Image.from_file` returns.

The end state we want to get here is to undo the usage of `tempfile.mktemp` and have examples that don't leave temporary files lying around after they exit. That should be possible by creating files long enough to write an image into them and read the data back. It looks like the same pattern is followed for every one of these examples. You might instead create the `Image` instance in the drawing functions which need the temporary file and then all the file handling can just be in one place.

Something like:
```python
def drawer():
    # ...
    with tempfile.NamedTemporaryFile(suffix='.bmp') as fid:
        gc.save(fid.name)
        image = Image.from_file(fid.name, resist_width='weak',
                                resist_height='weak')
    return image
```